python/google/protobuf/internal/type_checkers.py

```diff
@@ -51,6 +52,14 @@ def ToShortestFloat(original):
     rounded = float('{0:.{1}g}'.format(original, precision))
   return rounded
 
+def RoundTripsThroughDouble(x):
+  """ Returns true if x round-trips through being cast to a double, i.e., if
+  x is representable exactly as a double. This is a slightly weaker condition
+  than x < 2^52.
+  Represents the same behavior as src/google/protobuf/json/internal/unparser.cc
```

Comment:
Nit: Drop this comment reference now that its straightforward logic not ported from unparser.cc

Reply:
ack